src/NServiceBus.Core/Transports/Msmq/CheckEndpointNameComplianceForMsmq.cs

```diff
@@ -0,0 +1,17 @@
+﻿namespace NServiceBus.Transports.Msmq
+{
+    using System;
+
+    class CheckEndpointNameComplianceForMsmq
+    {
+        public void Check(string endpointName)
+        {
+            // .NET Messaging API hardcodes the buffer size to 124. As a result, the entire format name of the queue cannot exceed 123
+            var formatName = $"DIRECT=OS:{Environment.MachineName}\\private$\\{endpointName}";
+            if (formatName.Length > 123)
+            {
+                throw new InvalidOperationException($"The specified endpoint name {endpointName} is too long. The format name for the endpoint:{formatName} needs to be 123 characters or less.");
```

Comment:
If you want to know the endpoint name you could pass that in via `settings.EndpointName()`, or possibly the main shared queue via `settings.LocalAddress()` if you wanted an easy way to distinguish those values from the rest of the collection. Maybe adjust `Check` to take one of those and `bindings.ReceivingAddresses` directly?